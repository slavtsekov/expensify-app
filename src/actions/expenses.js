import database from "../firebase/firebase";

const addExpense = (expense) => ({
    type: "ADD_EXPENSE",
    expense
});

const requestAddExpense = (expenseData = {}) => {
    const { 
        description = "",
        note = "",
        amount = 0,
        createdAt = 0
    } = expenseData;
    const expense = { description, note, amount, createdAt };

    return {
        type: "REQUEST_ADD_EXPENSE",
        expense
    };
};

const removeExpense = ({ id }) => ({
    type: "REMOVE_EXPENSE",
    id
});

const requestRemoveExpense = ({ id }) => ({
    type: "REQUEST_REMOVE_EXPENSE",
    id 
});

const editExpense = (id, edited) => ({
    type: "EDIT_EXPENSE",
    id,
    edited
});

const requestEditExpense = (id, edited) => ({
    type: "REQUEST_EDIT_EXPENSE",
    id,
    edited
});

const setExpenses = (expenses) => ({
    type: "SET_EXPENSES",
    expenses
});

const requestSetExpenses = (success) => ({
    type: "REQUEST_SET_EXPENSES",
    success
});

export { 
    addExpense, 
    requestAddExpense,
    removeExpense,
    requestRemoveExpense,
    editExpense,
    requestEditExpense,
    setExpenses,
    requestSetExpenses
};