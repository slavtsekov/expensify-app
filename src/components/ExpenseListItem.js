import React from "react";
import { Link } from "react-router-dom";
import moment from "moment";
import numeral from "numeral";

const ExpenseListItem = (props) => {
    const { id, description, amount, createdAt } = props.data;
    return (
        <div>
            <Link to={`/edit/${id}`}>{description}</Link>
            - 
            {numeral(amount/100).format("$0,0.00")} 
            - 
            {moment(createdAt).format("MMMM Do, YYYY")}
        </div>
    );
};

export default ExpenseListItem;