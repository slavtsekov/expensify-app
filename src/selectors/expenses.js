const getVisibleExpenses = (expenses, {text, sortBy, startDate, endDate}) => {
    return expenses.filter((expense) => {
        const startDateMatch = typeof startDate !== "number" || expense.createdAt >= startDate;
        const endDateMatch = typeof endDate !== "number" || expense.createdAt <= endDate;
        const textMatch = expense.description.toLowerCase().includes(text.toLowerCase());

        return startDateMatch && endDateMatch && textMatch;
    }).sort((a, b) => {
        if (sortBy === "date") {
            return b.createdAt > a.createdAt ? 1 : -1;
        } else if (sortBy === "amount") {
            return b.amount > a.amount ? 1 : -1;
        }
    });
};

export { getVisibleExpenses as default };