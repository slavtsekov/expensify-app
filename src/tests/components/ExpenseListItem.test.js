import React from "react";
import { shallow } from "enzyme";
import ExpenseListItem from "../../components/ExpenseListItem";
import expenses from "../fixtures/expenses";

test("should render ExpenseListItem with data", () => {
    const wrapper = shallow(<ExpenseListItem data={expenses[0]} />);
    expect(wrapper).toMatchSnapshot();
});