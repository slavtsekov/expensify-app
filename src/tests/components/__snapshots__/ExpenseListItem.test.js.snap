// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem with data 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/1"
  >
    Gum
  </Link>
  -
  $1.95
  -
  January 1st, 1970
</div>
`;
